### PR TITLE
docs - ANALYZE command - HLL statistics, incremental analyze

### PR DESCRIPTION
**Pull request by mkiyama**
also updates to related topics for partitioned table statistics.

Link to HTML format documents with major changes in GPDB doc review site

ANALYZE command
http://docs-gpdb-review-staging.cfapps.io/review/ref_guide/sql_commands/ANALYZE.html

Changes to ORCA information about root partitions.
http://docs-gpdb-review-staging.cfapps.io/review/admin_guide/query/topics/query-piv-opt-root-partition.html

Add slots 4,5 to pg_statistic slots table and update partitioned table stats information
http://docs-gpdb-review-staging.cfapps.io/review/admin_guide/intro/about_statistics.html

optimizer_analyze_root_partition GUC
http://docs-gpdb-review-staging.cfapps.io/review/ref_guide/config_params/guc-list.html#optimizer_analyze_root_partition

This will be backported to 5X_STABLE with these changes
--FULLSCAN keyword is not available
--Query to list leaf-parts with HLL stats is changed.
--stakind5 is not available
--stakind5 =98 is removed, and --stakind5 =99 is moved to stakind4.

### CODE REVIEW

File: gpdb-doc/dita/admin_guide/intro/about_statistics.xml [hunk 1]
```diff
@@ -175,12 +174,73 @@
...
+                      <li><codeph>staop</codeph> contains the equality operator appropriate to the
+                        element type. </li>
+                      <li><codeph>stavalues</codeph> contains the most common element values.</li>
+                      <li><codeph>stanumbers</codeph> contains common element frequencies. </li>
+                    </ul><p>Frequencies are measured as the fraction of non-null rows the element
+                      value appears in, not the frequency of all rows. Also, the values are sorted
+                      into the element type's default order (to support binary search for a
+                      particular value). Since this puts the minimum and maximum frequencies at
+                      unpredictable spots in <codeph>stanumbers</codeph>, there are two extra
+                      members of <codeph>stanumbers</codeph>, that hold copies of the minimum and
```
Comment (dyozie) [inline gpdb-doc/dita/admin_guide/intro/about_statistics.xml]: remove comma after "stanumbers,"

File: gpdb-doc/dita/admin_guide/intro/about_statistics.xml [hunk 2]
```diff
@@ -175,12 +174,73 @@
...
+                      maximum frequencies. Optionally, there can be a third extra member, that holds
```
Comment (dyozie) [inline gpdb-doc/dita/admin_guide/intro/about_statistics.xml]: remove comma after "member,"

File: gpdb-doc/dita/admin_guide/intro/about_statistics.xml [hunk 3]
```diff
@@ -175,12 +174,73 @@
...
+                        counts into <codeph><varname>M</varname>-1</codeph> bins of approximately
+                        equal population. The first of these is the minimum observed count, and the
+                        last the maximum.</li>
+                    </ul><p>For child leaf partitions of a partitioned table,
+                        <codeph>stakind5</codeph> can contain the values 98 or 99.<ul
+                        id="ul_atn_4qm_y2b">
+                        <li>99 - <i>hyperloglog slot</i> stores the
+                            <codeph>hyperloglog_counter</codeph> created for sampled data. The
+                            <codeph>hyperloglog_counter</codeph> data structure is converted into a
+                            <codeph>bytea</codeph> and stored in <codeph>stavalues4</codeph> slot of
```
Comment (dyozie) [inline gpdb-doc/dita/admin_guide/intro/about_statistics.xml]: "stored in stavalues4 slot" -> Seems like this should be "stored in a stavalues4 slot" or "stored in the stavalues4 slot".  Not sure which is accurate.

File: gpdb-doc/dita/ref_guide/sql_commands/ANALYZE.xml [hunk 4]
```diff
@@ -100,46 +140,66 @@
...
+      <p>If you run <codeph>ANALYZE</codeph> on a table that does not contain data, statistics are
+        not collected for the table. For example, if you perform a <codeph>TRUNCATE</codeph>
+        operation on a table that has statistics, and then run <codeph>ANALYZE</codeph> on the
+        table, the statistics do not change.</p>
       <p>For a partitioned table, specifying which portion of the table to analyze, the root
-        partition or subpartitions (leaf child tables) can be useful if the partitioned table has
-        large number of partitions that have been analyzed and only a few leaf child tables have
-        changed. </p>
+        partition or subpartitions (leaf child partition tables) can be useful if the partitioned
+        table has large number of partitions that have been analyzed and only a few leaf child
```
Comment (dyozie) [inline gpdb-doc/dita/ref_guide/sql_commands/ANALYZE.xml]: has large -> has a large

File: gpdb-doc/dita/ref_guide/sql_commands/ANALYZE.xml [hunk 5]
```diff
@@ -100,46 +140,66 @@
...
-            with the <codeph>CREATE TABLE</codeph> command, Greenplum Database creates the table
-            that you specify (the root partition or parent table), and also creates a hierarchy of
-            tables based on the partition hierarchy that you specified (the child tables).
-            Partitioned tables, child tables and their inheritance level relationships are tracked
-            in the system view <i>pg_partitions</i>.</note></li>
+        <li>Running <codeph>ANALYZE FULLSCAN</codeph> collects more accurate statistics than running
+            <codeph>ANALYZE</codeph>, but requires more time.</li>
+        <li id="bd138200">When you run <codeph>ANALYZE [FULLSCAN]</codeph> on the root partitioned
+          table, statistics are collected for all the leaf child partitions. Leaf child partitions
+          are lowest-level tables in the hierarchy of child tables created by Greenplum Database for
```
Comment (dyozie) [inline gpdb-doc/dita/ref_guide/sql_commands/ANALYZE.xml]: are lowest -> are the lowest

File: gpdb-doc/dita/ref_guide/sql_commands/ANALYZE.xml [hunk 6]
```diff
@@ -100,46 +140,66 @@
...
+            collected. If you run <codeph>ANALYZE</codeph> on the leaf child partition for March of
+            2013, statistics are collected only for that leaf child partition.</p><p>When you run
+              <codeph>ANALYZE FULLSCAN</codeph> on a leaf child partition, <codeph>ANALYZE</codeph>
+            updates the root partition statistics when possible.</p><note>When you create a
+            partitioned table with the <codeph>CREATE TABLE</codeph> command, Greenplum Database
+            creates the table that you specify (the root partition or parent table), and also
+            creates a hierarchy of tables based on the partition hierarchy that you specified (the
+            child tables). Partitioned tables, child tables and their inheritance level
+            relationships are tracked in the system view <i>pg_partitions</i>.</note></li>
+        <li>For leaf child partition that has changed, for example, you exchange the leaf child
```
Comment (dyozie) [inline gpdb-doc/dita/ref_guide/sql_commands/ANALYZE.xml]: For leaf -> For a leaf

Also recommend editing and putting the parenthetical in parens:  (for example, if you exchanged the leaf child partition or made significant updates to the leaf child partition data)

**Comment by mkiyama** [comment]
closing. Will open a new PR with features based on dev email.

**Pull request closed without merging**